nio/client/async_client.py

```diff
@@ -534,7 +537,18 @@ async def create_matrix_response(
             resp = response_class.from_data(parsed_dict, content_type, name)
 
         elif issubclass(response_class, FileResponse):
-            body = await transport_response.read()
+            if not save_to:
+                body = await transport_response.read()
+            else:
+                if os.path.isdir(save_to):
+                    save_to = os.path.join(save_to, name)
+                async with aiofiles.open(save_to, "wb") as f:
+                    while True:
+                        chunk = await transport_response.content.read(1024 * 4096)
```

Comment:
Unrelated to other design considerations: these values seem like the sort of thing that should be deferred to [AsyncClientConfig](https://github.com/poljar/matrix-nio/blob/1db796d84fa3d3e91ce9b8e21559d73938e0423a/nio/client/async_client.py#L312)